Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -125,4 +128,37 @@ Vector<String> transcodeImages(const Vector<String>& paths, const String& destin
     });
 }
 
+std::pair<String, Vector<IntSize>> utiAndAvailableSizesFromImageData(std::span<const uint8_t> data)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
```

Comment:
In case you want to differentiate between the case of decoding error and unsupported image types, you can check for the encodedDataStatus() here:
```
if (imageDecoder->encodedDataStatus() == EncodedDataStatus::Error)
    // return a decoding error here
```

Reply:
Added error.